core/src/components/nav/nav.tsx

```diff
@@ -607,25 +579,33 @@ export class Nav implements NavOutlet {
     }
   }
 
+  /**
+   * Consumes the next transition in the queue.
+   *
+   * @returns whether the transition is executed.
+   */
   private nextTrns(): boolean {
     // this is the framework's bread 'n butta function
     // only one transition is allowed at any given time
     if (this.isTransitioning) {
       return false;
     }
 
-    // there is no transition happening right now
-    // get the next instruction
+    // there is no transition happening right now, executes the next instructions.
     const ti = this.transInstr.shift();
     if (!ti) {
       return false;
     }
-
-    this.runTransition(ti);
+    this.runTransitions(ti);
     return true;
   }
 
-  private async runTransition(ti: TransitionInstruction) {
+  /**
+   * Executes all the transition instruction from the queue.
+   *
+   * @param ti The transition instructions.
+   */
+  private async runTransitions(ti: TransitionInstruction) {
```

Comment:
Any particular reason for this change? This function does run a single transition, so maybe keeping the singular form would be best. I realize it does call `nextTrns` which causes this to be run again with another transition, but the body of this function only deals with a single transition at a time.

Reply:
Agreed. I changed the name when I was looking how transitions were chained but the singular is probably ad good.